sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java

```diff
@@ -99,7 +102,15 @@ public Mono<Void> initializeEncryptionSettingsAsync(boolean isRetry) {
         {
             this.containerRid = cosmosContainerProperties.getResourceId();
             this.databaseRid = cosmosContainerPropertiesAccessor.getSelfLink(cosmosContainerProperties).split("/")[1];
+
+            if (cosmosContainerProperties.getPartitionKeyDefinition().getPaths().isEmpty()) {
```

Comment:
i think the if/else is siwtched around? If so, means missing test coverage

Reply:
That if statement should be that the getPaths() is not empty. I have updated the code accordingly.